pkg/trace/operation.go

```diff
@@ -201,20 +203,28 @@ func FromOperation(parent Operation, format string, args ...interface{}) Operati
 	return parent.newChild(parent.Context, fmt.Sprintf(format, args...))
 }
 
-// FromContext unpacks the values in the ctx to create an Operation
-func FromContext(ctx context.Context) (Operation, error) {
-
-	o := Operation{
-		Context: ctx,
+// FromContext will return an Operation
+//
+// The Operation returned will be one of the following:
+//   The operation in the context value
+//   The operation passed as the context param
+//   A new operation
+func FromContext(ctx context.Context, message string) Operation {
+
+	// do we have a context w/the op added as a value
+	if op, ok := ctx.Value(OpTraceKey).(Operation); ok {
```

Comment:
can we have an extra sanity check in the condition `op != nil`. Stupid things can happen and while this really should result in a panic we also need to continue functioning.

Reply:
`op` could never == nil in this case.  I guess it could be the zero value of the operation struct, so I've added a check to ensure there's an `op.id` -- so if we have a zero value we will return a new operation otherwise an operation based on the one passed via `context.Value`